iggy/src/cmd/personal_access_tokens/get_personal_access_tokens.rs

```diff
@@ -0,0 +1,86 @@
+use crate::cli_command::{CliCommand, PRINT_TARGET};
+use crate::client::Client;
+use crate::personal_access_tokens::get_personal_access_tokens::GetPersonalAccessTokens;
+use anyhow::Context;
+use async_trait::async_trait;
+use chrono::{DateTime, Local};
+use comfy_table::Table;
+use std::time::{Duration, UNIX_EPOCH};
+use tracing::{event, Level};
+
+pub enum GetPersonalAccessTokensOutput {
+    Table,
+    List,
+}
+
+pub struct GetPersonalAccessTokensCmd {
+    get_tokens: GetPersonalAccessTokens,
+    output: GetPersonalAccessTokensOutput,
+}
+
+impl GetPersonalAccessTokensCmd {
+    pub fn new(output: GetPersonalAccessTokensOutput) -> Self {
+        Self {
+            get_tokens: GetPersonalAccessTokens {},
+            output,
+        }
+    }
+}
+
+#[async_trait]
+impl CliCommand for GetPersonalAccessTokensCmd {
+    fn explain(&self) -> String {
+        let mode = match self.output {
+            GetPersonalAccessTokensOutput::Table => "table",
+            GetPersonalAccessTokensOutput::List => "list",
+        };
+        format!("list personal access tokens in {mode} mode")
+    }
+
+    async fn execute_cmd(&mut self, client: &dyn Client) -> anyhow::Result<(), anyhow::Error> {
+        let tokens = client
+            .get_personal_access_tokens(&self.get_tokens)
+            .await
+            .with_context(|| String::from("Problem getting list of personal access tokens"))?;
+
+        match self.output {
+            GetPersonalAccessTokensOutput::Table => {
+                let mut table = Table::new();
+
+                table.set_header(vec!["Name", "Token Expiry Time"]);
+
+                tokens.iter().for_each(|token| {
+                    table.add_row(vec![
+                        format!("{}", token.name.clone()),
+                        match token.expiry {
+                            Some(value) => {
+                                DateTime::<Local>::from(UNIX_EPOCH + Duration::from_micros(value))
```

Comment:
When listing the tokens, the returned `expiry` is already the actual timestamp based on epoch. Only when creating the tokens, the `expiry` is the amount of the time to be added to the current timestamp generated on the server.

```rust
let expiry = expiry.map(|e| now + e as u64 * 1_000_000);
```

Reply:
Ok, changed to use `TimeStamp` and added new method `to_local` to `TimeStamp` in order to be able to render local time stamp instead of UTC.